scripts/correlation_test.py

```diff
@@ -106,13 +117,74 @@ def get_random_score_ratings(user_ratings_list, num):
         ans.append(permute_dict)
     return ans
 
+def reverse(edge):
+    start = edge[0]
+    end = edge[1]
+    result = (end, start)
+    return result
+
+def get_random_edges(user_ratings, graph, num_permutations):
+    '''
+    Takes in user ratings for a particular restaurant, the friend graph, and the number
+    randomly generated permutations to create
+    Return a list of user->list of friends dictionaries
+    '''
+
+    #create the original set of edges
+    edges = []
+    for user in user_ratings:
+        start = user[0]         
+        for friend in graph[start]:
+            e = (start, friend)
+            edges.append(e)
+
+    #the list of randomized adjacency lists to return
+    randomized_graphs = []
+
+    for i in range(num_permutations):
+        randomized_edges = list(edges)
+        #pick two random edges from the list and swap their endpoints
+        for i in range(len(edges)):
+            first = random.choice(randomized_edges)
+            second = random.choice(randomized_edges)
+            while second is first:
```

Comment:
hmm I wonder if this is causing the slowdown. If there aren't many edges this might be incredibly slow? But it seems like since you're doing this for the whole graph it should be fine. I think the fact that you're trying to randomize the entire graph might be incredibly slow, so it might be worth just filtering to only the adj lists you need?